x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRestoreSourceService.java

```diff
@@ -133,32 +140,73 @@ public synchronized void closeSession(String sessionUUID) {
         IOUtils.closeWhileHandlingException(restore);
     }
 
+    public synchronized Reader getSessionReader(String sessionUUID, String fileName) throws IOException {
+        RestoreContext restore = onGoingRestores.get(sessionUUID);
+        if (restore == null) {
+            logger.info("could not get session [{}] because session not found", sessionUUID);
```

Comment:
do we need to log at info level here? Throwing the exception will fail the recovery and bubble up the failure already

Reply:
Changed.